src/main/kotlin/org/rust/ide/intentions/visibility/ChangeVisibilityIntention.kt

```diff
@@ -0,0 +1,102 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.ide.intentions.visibility
+
+import com.intellij.openapi.editor.Editor
+import com.intellij.openapi.project.Project
+import com.intellij.psi.PsiElement
+import org.rust.cargo.project.workspace.PackageOrigin
+import org.rust.ide.intentions.RsElementBaseIntentionAction
+import org.rust.lang.core.psi.*
+import org.rust.lang.core.psi.ext.*
+
+abstract class ChangeVisibilityIntention : RsElementBaseIntentionAction<ChangeVisibilityIntention.Context>() {
+    override fun getFamilyName(): String = "Change item visibility"
+
+    abstract val visibility: String
+    abstract fun isApplicable(element: RsVisibilityOwner): Boolean
+
+    data class Context(val element: RsVisibilityOwner)
+
+    override fun findApplicableContext(project: Project, editor: Editor, element: PsiElement): Context? {
+        val visibleElement = element.ancestorStrict<RsVisibilityOwner>() ?: return null
```

Comment:
Oh, I forget one more comment. Let's restrict the intention range to function modifiers (just like in Kotlin) and use `checkAvailableInSelectionOnly` to test it